### PR TITLE
[WIP] refactor: deployment service nestjs

### PR DESCRIPTION
**Pull request by bashleigh**
# Pull request checklist

**Detail as per issue below (required):**

fixes: #

### CODE REVIEW

**Review by joshbalfour — DISMISSED** [review]

File: packages/deployment-service/.env.dist [hunk 1]
```diff
@@ -0,0 +1,4 @@
+MYSQL_ROOT_PASSWORD=root
```
Comment (willmcvay) [inline packages/deployment-service/.env.dist:1]: Erm, I think this should be gitignored. And probably best to change the PW

File: packages/deployment-service/src/default-header-interceptor.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { CallHandler, ExecutionContext, Injectable, NestInterceptor } from '@nestjs/common'
+import { Observable } from 'rxjs'
+import { Response } from 'express'
+
+@Injectable()
+export class DefaultHeaderInterceptor implements NestInterceptor {
+  intercept(context: ExecutionContext, next: CallHandler): Observable<any> {
+    const response = context.switchToHttp().getResponse<Response>()
+
+    response.setHeader('Access-Control-Allow-Origin', '*')
```
Comment (willmcvay) [inline packages/deployment-service/src/default-header-interceptor.ts:10]: Think you can just use the cors package to fix this in Sonarcloud

**Review by willmcvay — CHANGES_REQUESTED** [review]
I've had a good look but TBH I'm not familiar enough with Nest to offer any meaningful feedback on the code, suggest maybe asking @joshbalfour to check it out as well before merging. Please can we fix the Sonarcloud issues and that DB creds leak too

File: packages/deployment-service/.env.dist (hunk 1 above)
Comment (bashleigh) [reply packages/deployment-service/.env.dist:1]: This is fine, it's a dist so the idea is to copy this from `.env.dist` to `.env` and use `.env` for dev

File: packages/deployment-service/src/default-header-interceptor.ts (hunk 2 above)
Comment (bashleigh) [reply packages/deployment-service/src/default-header-interceptor.ts:10]: I'd rather do it like this than to add another package

**Review by willmcvay — APPROVED** [review]

**Pull request merged**